update autogen.sh and Rx xml addition

rakeshgmtke opened the pull request:
1. Update Autogen.sh scipt to handle new line with echo
2. Rx xml added to and generated avp/codes/application file update with autogen.sh script

rakeshgmtke committed:
update autogen.sh

Update Autogen.sh scipt to handle new line with echo

echo cmd is updated with -e to avoid error adding new line.

-e     enable interpretation of backslash escapes

echo manual:
       If -e is in effect, the following sequences are recognized:

       \\     backslash

       \a     alert (BEL)

       \b     backspace

       \c     produce no further output

       \e     escape

       \f     form feed

       \n     new line

       \r     carriage return

       \t     horizontal tab

       \v     vertical tab

       \0NNN  byte with octal value NNN (1 to 3 digits)

       \xHH   byte with hexadecimal value HH (1 to 2 digits)

       NOTE: your shell may have its own version of echo, which usually supersedes the version described here.  Please refer to your shell's documentation for details about the options
       it supports.

rakeshgmtke committed:
Rx xml addition

Rx xml added to and generated avp/codes/application file update with autogen.sh script

default.go is added Rx xml from testdata/tgpp_rx.xml

rakeshgmtke commented:
Above changes can fix issue **https://github.com/fiorix/go-diameter/issues/208**

rakeshgmtke committed:
Adding Cx interface support v1

Adding Cx interface support

rakeshgmtke committed:
Adding Cx interface support v2

Adding Cx interface support v2

handling UAR/MAR/LIR/SAR

rakeshgmtke committed:
Revert "Adding Cx interface support v2"

This reverts commit 11d1ca5e53cad5145780af018ee0984e7a917ca2.

rakeshgmtke committed:
changing GitHub reference rakeshgmtke

fiorix --> rakeshgmtke

rakeshgmtke committed:
Update go.mod

rakeshgmtke committed:
Update go.mod

rakeshgmtke committed:
Update go.mod

rakeshgmtke committed:
Update go.mod

fiorix commented:
This cannot be merged until you update the package path back to the original. All of your code is now using your own clone and this change would break this repo.

rakeshgmtke committed:
Modify tgpp_cx.xml with updated commands and AVPs

Updated command URLs and added new AVPs for Push-Profile.

rakeshgmtke committed:
Remove obsolete User-Data-Request-Type AVP

Removed obsolete User-Data-Request-Type AVP from XML.

rakeshgmtke committed:
Update command links and add AVPs in default.go

Updated command documentation links and added missing AVPs for Push-Profile command.

rakeshgmtke committed:
Fix comment formatting in tgpp_cx.xml

rakeshgmtke committed:
Fix comment formatting in default.go

fiorix commented:
Thanks @rakeshgmtke for the effort on adding Rx and Cx dictionary support — that's valuable work!

Unfortunately this PR can't be merged as-is due to several issues:

1. **Module path changed to your fork** — all imports point to `github.com/rakeshgmtke/go-diameter` instead of `github.com/fiorix/go-diameter/v4`, which would break the module for all users
2. **Duplicate constants** — `MultimediaAuth` conflicts with existing `MultimediaAuthentication` (both = 303), and `FramedIpv6Prefix` conflicts with `FramedIPv6Prefix` (both = 97)
3. **`echo -e` portability** — the script uses `#!/bin/sh` but `echo -e` is a bashism; should use `printf`
4. **Large diff from reordering** — regenerated files reorder existing entries making review difficult

If you'd like to rework this, a fresh PR with the correct module path and the above issues addressed would be welcome. Thanks again for the contribution!

dunst0 commented:
@fiorix Quick question since I'm just checking this PR here as I would like to use this library for some little project.

From reading the specs for SWx and Cx/Dx I would say the name for the command would be better MultimediaAuth than MultimediaAuthentication.

Cx/Dx interface description from 3GPP TS 29.229
```
< Multimedia-Auth-Request > ::= < Diameter Header: 303, REQ, PXY, 16777216 >
                                < Session-Id >
                                { Vendor-Specific-Application-Id }
                                { Auth-Session-State }
                                { Origin-Host }
                                { Origin-Realm }
                                { Destination-Realm
```

SWx interface description from 3GPP TS 29.273
```
< Multimedia-Auth-Request > ::= < Diameter Header: 303, REQ, PXY, 16777265 >
                                < Session-Id >
                                [ DRMP ]
                                { Vendor-Specific-Application-Id }
                                { Auth-Session-State }
                                { Origin-Host }
                                { Origin-Realm } 
```

Would it be an option to rename the old one or is it now set in stone?

fiorix commented:
Good catch @dunst0, you're right — the 3GPP specs (TS 29.229, TS 29.273) define it as Multimedia-Auth-Request/Answer (MAR/MAA), not Multimedia-Authentication.

Fixed in ce954f4 — renamed the SWx dictionary entry to `Multimedia-Auth` and the Go constant to `MultimediaAuth`. A deprecated `MultimediaAuthentication` alias is kept for backward compatibility.
